services/elasticache/pkg/resource/replication_group/custom_update_api.go

```diff
@@ -300,13 +344,35 @@ func (rm *resourceManager) newDecreaseReplicaCountRequestPayload(
 		res.SetNewReplicaCount(*desiredSpec.ReplicasPerNodeGroup)
 	}
 
+	latestStatus := latest.ko.Status
+	// each shard could have different value for replica count
+	latestReplicaCounts := map[string]int{}
+	for _, latestShard := range latestStatus.NodeGroups {
+		if latestShard.NodeGroupID == nil {
+			continue
+		}
+		latestReplicaCount := 0
+		if latestShard.NodeGroupMembers != nil {
+			if len(latestShard.NodeGroupMembers) > 0 {
+				latestReplicaCount = len(latestShard.NodeGroupMembers) - 1
+			}
+		}
+		latestReplicaCounts[*latestShard.NodeGroupID] = latestReplicaCount
+	}
```

Comment:
suggestion for a future cleanup patch... you could make a little helper function that has the above code so you don't need to repeat it from `newIncreaseReplicaCountRequestPayload`.